src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsUrlTest.java

```diff
@@ -0,0 +1,201 @@
+                });
+                checksNamesMap.put(dirName, fileNames);
+            }
+            final String[] miscNames = checks.list((rootDir, pathname) -> {
+                return pathname.endsWith(SUFFIX_CHECK_JAVA);
+            });
+            checksNamesMap.put(MISC, miscNames);
+        }
+        return checksNamesMap;
+    }
+
+    @Test
+    public void testXdocsUrl() throws Exception {
+        final SAXParserFactory parserFactory = SAXParserFactory.newInstance();
+        final SAXParser parser = parserFactory.newSAXParser();
+        final CheckTest checkHandler = new CheckTest(NODE_NAME);
+        final String[][] strings;
+        try (InputStream input = Files.newInputStream(
+                Paths.get(getRootPath() + "/" + getXmlPath()))) {
+            parser.parse(input, checkHandler);
+            strings = checkHandler.getStrings();
+        }
+        final Map<String, String[]> checksNamesMap = getXdocsMap();
+        for (String[] sub : strings) {
+            assertEquals(sub[1], sub[2],
+                    "URL in" + getXmlPath() + " is inconsistent: "
+                            + sub[0] + " " + sub[1] + " " + sub[2]);
+            if (sub[1].equals(COMMENTS_INDENTATION) || sub[1].equals(INDENTATION)) {
+                assertEquals(MISC, sub[0],
+                        "URL in" + getXmlPath() + " is inconsistent: "
+                                + sub[0] + " " + sub[1] + " " + sub[2]);
+            }
+            else if (sub[1].equals(SUPPRESS_WARNINGS_HOLDER)) {
+                assertEquals(ANNOTATION, sub[0],
+                        "URL in" + getXmlPath() + " is inconsistent: "
+                                + sub[0] + " " + sub[1] + " " + sub[2]);
+            }
+            else {
+                final String[] moduleFileNames = checksNamesMap.get(sub[0]);
+                if (moduleFileNames != null) {
+                    boolean match = false;
+                    for (String filename : moduleFileNames) {
+                        if ((sub[1] + SUFFIX_CHECK_JAVA).equals(filename)) {
+                            match = true;
+                            break;
+                        }
+                    }
+                    assertTrue(match, "URL in" + getXmlPath() + " is inconsistent: "
+                            + sub[0] + " " + sub[1] + " " + sub[2]);
+                }
+            }
+        }
+    }
+
+    public static final class CheckTest extends DefaultHandler {
+
+        public static final String SPLIT_ONE = "#";
+
+        public static final String SPLIT_TWO = "\\.";
+
+        public static final String PREFIX_CONFIG = "config_";
+
+        private final String nodeName;
+
+        private String[] strings;
+
+        private List<String[]> list;
+
+        private String currentTag;
+
+        public CheckTest(String nodeName) {
+            this.nodeName = nodeName;
+        }
+
+        public String[][] getStrings() {
```

Comment:
Why don't use list or map? it is way more convenient than multidimensional arrays. Please change.

Reply:
done~